### ISSUE TITLE
Any plan to support instance normalization?

### ISSUE THREAD

**Issue by heagoo** [intel/mkl-dnn#263]
This is not a bug, but a feature request.

Just want to know, is there any plan to support instance normalization?

If it can not be available in near future, we will implement it ourselves (but just for inference). 

Thanks!

**Comment by emfomenk** [comment]
Hi @heagoo,

We are considering to extend our batch-normalization to process some group of images independently, e.g. for batch size 128 you might be able to perform 4 independent batch normalizations (each of size 32). We don't have a name for that yet, maybe grouped-bnorm. But anyways, instant normalization is just a particular case -- the number of groups is equal to the batch size.

Alas, no ETA so far. Maybe @vpirogov can comment a little more on that.

Could you please share some details on where is this primitive/approach used or where do you plan it to use?

**Comment by emfomenk** [comment]
Implementation wise (if you would do that yourself) -- as i said earlier from our perspective it makes sense to reuse batch-normalization interface, e.g. by adding a *number of groups* parameter. 

But even if you decide to go with a standalone primitive [the following comment](https://github.com/intel/mkl-dnn/issues/241#issuecomment-390702503) in neighboring thread might be useful.

**Comment by heagoo** [comment]
@emfomenk We are using MKLDNN to help to accelerate the inference of an AI algorithm, which is used to recognize white and black pieces in Go chess (from a picture).

Thanks for the reminding to reuse BatchNorm, that should work. And, as current batch size = 1, I think we can directly use BatchNorm without any modification.
